### ISSUE TITLE
The "entity_browser_widget_context" plugin does not exist.

### ISSUE THREAD

**Issue by gabe-connolly** [acquia/lightning#377]
My system information:
Operating system type: OSX
Operating system version: 10.12.4
BLT version: 8.8.3

When I run this command:
```
drush @mysite.local site-install 
```

I get the following output:
```
You are about to DROP all tables in your 'drupal' database. Do you want to continue? (y/n): y
Starting Drupal installation. This takes a while. Consider using the --notify global option.                                 [ok]
Drupal\Component\Plugin\Exception\PluginNotFoundException: The "entity_browser_widget_context" plugin does not exist.     [error]
in /var/www/mysite/docroot/core/lib/Drupal/Component/Plugin/Discovery/DiscoveryTrait.php:52
Stack trace:
#0 /var/www/mysite/docroot/core/lib/Drupal/Component/Plugin/Discovery/DiscoveryCachedTrait.php(25):
Drupal\Core\Plugin\DefaultPluginManager->doGetDefinition(Array, 'entity_browser_...', true)
#1 /var/www/mysite/docroot/core/lib/Drupal/Core/Plugin/Factory/ContainerFactory.php(16):
Drupal\Core\Plugin\DefaultPluginManager->getDefinition('entity_browser_...')
#2 /var/www/mysite/docroot/core/lib/Drupal/Component/Plugin/PluginManagerBase.php(84):
Drupal\Core\Plugin\Factory\ContainerFactory->createInstance('entity_browser_...', Array)
#3 /var/www/mysite/docroot/core/modules/views/src/Plugin/views/argument/ArgumentPluginBase.php(1111):
Drupal\Component\Plugin\PluginManagerBase->createInstance('entity_browser_...')
#4 /var/www/mysite/docroot/core/modules/views/src/Plugin/views/argument/ArgumentPluginBase.php(1316):
Drupal\views\Plugin\views\argument\ArgumentPluginBase->getPlugin('argument_defaul...')
#5 /var/www/mysite/docroot/core/lib/Drupal/Core/Plugin/PluginDependencyTrait.php(47):
Drupal\views\Plugin\views\argument\ArgumentPluginBase->calculateDependencies()
#6 [internal function]:
Drupal\views\Plugin\views\display\DisplayPluginBase->calculatePluginDependencies(Object(Drupal\views\Plugin\views\argument\StringArgument),
2)
#7 /var/www/mysite/docroot/core/modules/views/src/Plugin/views/display/DisplayPluginBase.php(953):
array_walk(Array, Array)
#8 /var/www/mysite/docroot/core/lib/Drupal/Core/Plugin/PluginDependencyTrait.php(47):
Drupal\views\Plugin\views\display\DisplayPluginBase->calculateDependencies()
#9 /var/www/mysite/docroot/core/modules/views/src/Entity/View.php(281):
Drupal\Core\Config\Entity\ConfigEntityBase->calculatePluginDependencies(Object(Drupal\entity_browser\Plugin\views\display\EntityBrowser))
#10 /var/www/mysite/docroot/core/lib/Drupal/Core/Config/Entity/ConfigEntityBase.php(346):
Drupal\views\Entity\View->calculateDependencies()
#11 /var/www/mysite/docroot/core/modules/views/src/Entity/View.php(291):
Drupal\Core\Config\Entity\ConfigEntityBase->preSave(Object(Drupal\Core\Config\Entity\ConfigEntityStorage))
#12 /var/www/mysite/docroot/core/lib/Drupal/Core/Entity/EntityStorageBase.php(434):
Drupal\views\Entity\View->preSave(Object(Drupal\Core\Config\Entity\ConfigEntityStorage))
#13 /var/www/mysite/docroot/core/lib/Drupal/Core/Entity/EntityStorageBase.php(389):
Drupal\Core\Entity\EntityStorageBase->doPreSave(Object(Drupal\views\Entity\View))
#14 /var/www/mysite/docroot/core/lib/Drupal/Core/Config/Entity/ConfigEntityStorage.php(259):
Drupal\Core\Entity\EntityStorageBase->save(Object(Drupal\views\Entity\View))
#15 /var/www/mysite/docroot/core/lib/Drupal/Core/Entity/Entity.php(364):
Drupal\Core\Config\Entity\ConfigEntityStorage->save(Object(Drupal\views\Entity\View))
#16 /var/www/mysite/docroot/core/lib/Drupal/Core/Config/Entity/ConfigEntityBase.php(637):
Drupal\Core\Entity\Entity->save()
#17
/var/www/mysite/docroot/profiles/contrib/lightning/profiles/lightning/modules/lightning_features/lightning_media/lightning_media.install(151):
Drupal\Core\Config\Entity\ConfigEntityBase->save()
#18
/var/www/mysite/docroot/profiles/contrib/lightning/profiles/lightning/modules/lightning_features/lightning_media/lightning_media.install(31):
lightning_media_update_8008()
#19 [internal function]: lightning_media_install()
#20 /var/www/mysite/docroot/core/lib/Drupal/Core/Extension/ModuleHandler.php(391):
call_user_func_array('lightning_media...', Array)
#21 /var/www/mysite/docroot/core/lib/Drupal/Core/Extension/ModuleInstaller.php(287):
Drupal\Core\Extension\ModuleHandler->invoke('lightning_media', 'install')
#22 /var/www/mysite/docroot/core/lib/Drupal/Core/ProxyClass/Extension/ModuleInstaller.php(83):
Drupal\Core\Extension\ModuleInstaller->install(Array, false)
#23 /var/www/mysite/docroot/core/includes/install.core.inc(1808):
Drupal\Core\ProxyClass\Extension\ModuleInstaller->install(Array, false)
#24 /var/www/mysite/docroot/core/includes/batch.inc(252): _install_module_batch('lightning_media', 'Lightning
Media', Array)
#25 /var/www/mysite/docroot/core/includes/form.inc(874): _batch_process()
#26 /var/www/mysite/docroot/core/includes/install.core.inc(625): batch_process(Object(Drupal\Core\Url),
Object(Drupal\Core\Url))
#27 /var/www/mysite/docroot/core/includes/install.core.inc(546): install_run_task(Array, Array)
#28 /var/www/mysite/docroot/core/includes/install.core.inc(117): install_run_tasks(Array)
#29 /var/www/mysite/vendor/drush/drush/includes/drush.inc(726):
install_drupal(Object(Composer\Autoload\ClassLoader), Array)
#30 /var/www/mysite/vendor/drush/drush/includes/drush.inc(711): drush_call_user_func_array('install_drupal',
Array)
#31 /var/www/mysite/vendor/drush/drush/commands/core/drupal/site_install.inc(84): drush_op('install_drupal',
Object(Composer\Autoload\ClassLoader), Array)
#32 /var/www/mysite/vendor/drush/drush/commands/core/site_install.drush.inc(254):
drush_core_site_install_version('lightning', Array)
#33 /var/www/mysite/vendor/drush/drush/includes/command.inc(422): drush_core_site_install()
#34 /var/www/mysite/vendor/drush/drush/includes/command.inc(231): _drush_invoke_hooks(Array, Array)
#35 /var/www/mysite/vendor/drush/drush/includes/command.inc(199): drush_command()
#36 /var/www/mysite/vendor/drush/drush/lib/Drush/Boot/BaseBoot.php(67): drush_dispatch(Array)
#37 /var/www/mysite/vendor/drush/drush/includes/preflight.inc(66):
Drush\Boot\BaseBoot->bootstrap_and_dispatch()
#38 /var/www/mysite/vendor/drush/drush/drush.php(12): drush_main()
#39 {main}
```

And I expected this to happen:
I expect the a fresh base Drupal install with no errors.

I am building Lightning via composer: `"drupal/lightning": "^2.13",`

In trying to troubleshoot this issue, I have traced the problem to docroot/profiles/contrib/lightning/profiles/lightning/modules/lightning_features/lightning_media/lightning_media.install

Specifically, the contents of this are causing my install to fail: `$display = array_merge($display, unserialize(...)`

I was able to bypass the site-install error by unserializing the content from the above code, removing the following line, and then re-serializing the content: 
`'default_argument_type' => 'entity_browser_widget_context',`

This is not a sustainable solution however, since it requires modifying Lightning core files.

**Comment by phenaproxima** [comment]
@mortenson, do you have any input on this?

**Comment by mortenson** [comment]
@phenaproxima I'm guessing the patch from https://github.com/acquia/lightning/blob/8.x-2.x/composer.json#L106 didn't get applied properly when you installed Lightning.

@gabe-connolly Could you try running `composer clear-cache` and re-installing Lightning with composer?

**Comment by gabe-connolly** [comment]
@mortenson I have tried completely removing my local site and starting from scratch, but still with no luck.  I am using BLT for this project, so this was my workflow for the reinstall:

- `composer clear-cache`
- `composer install`
- `blt vm`
- `blt local:setup`

At this point I'm back to getting the following error:

> Drupal\Component\Plugin\Exception\PluginNotFoundException: The "entity_browser_widget_context" plugin does not exist. in                                             [error]
> /var/www/mysite/docroot/core/lib/Drupal/Component/Plugin/Discovery/DiscoveryTrait.php:52

Before destroying the site, I was able to complete a `blt setup` after correcting some issues with our config-split files.  Now though, I'm stuck again before my config setting can be applied because of the missing plugin error.

**Comment by gabe-connolly** [comment]
@mortenson update: I added the lightning patch you recommended directly to my project's composer.json file, and that seemed to do the trick:

For anyone else having similar issues, here is the patch block I ended up using for drupal/entity_browser

```
"drupal/entity_browser": {
                "2865928 - The View widget should filter based on field settings":
                "https://www.drupal.org/files/issues/entity-browser-view-context-2865928-8.patch",
                "Error removing image added with previous version":
                "https://www.drupal.org/files/issues/entity_browser-8.x-1.0-2820132-17.patch"
            },
```

**Comment by balsama** [comment]
Just want to point out that you have another problem if that patch isn't getting applied properly without that in your composer.json file.

composer-patches can be finicky. I would try to get to the bottom of it now before it starts causing problems later on. BLT has a composer nuke command that you might want to try.

**Comment by gabe-connolly** [comment]
Hm, thanks @balsama .  I'll keep digging for the root cause.  At this point, I suspect we have something in our config import that is causing the issue, but I'm having a hard time narrowing it down from there.  [This issue](https://github.com/acquia/blt/issues/1470) on the BLT project seems to pretty closely mirror the issues I've been having, but unfortunately it's not clear how that issue config was finally resolved there.

**Comment by gabe-connolly** [comment]
So, after a lot of trial and error I believe I have narrowed down the remaining configuration import issues to a conflict with the lightning install profile.  Specifically, when I run a `blt setup` now, I get all the way through the install process and configuration import only to end up with this failure message:

> Finalizing configuration synchronization.                                                                                                                               [ok]
> The import failed due for the following reasons:                                                                                                                     [error]
> Unexpected error during import with operation delete for media_entity.bundle.tweet: Attempt to create a field body that does not exist on entity type node.
> Unexpected error during import with operation delete for media_entity.bundle.instagram: Attempt to create a field body that does not exist on entity type node.
> Unexpected error during import with operation delete for media_entity.bundle.image: Attempt to create a field body that does not exist on entity type node.
> Unexpected error during import with operation delete for media_entity.bundle.document: Attempt to create a field body that does not exist on entity type node.
> Unexpected error during import with operation delete for contact.form.personal: Attempt to create a field body that does not exist on entity type node.
> Unexpected error during import with operation delete for block_content.type.basic: Attempt to create a field body that does not exist on entity type node.
> Unexpected error during import with operation delete for field.field.node.landing_page.body: Attempt to create a field body that does not exist on entity type
> node.
> Unexpected error during import with operation delete for field.field.node.landing_page.field_meta_tags: Attempt to create a field field_meta_tags that does not
> exist on entity type node.
> Unexpected error during import with operation delete for node.type.landing_page: Attempt to create a field body that does not exist on entity type node.
> Unexpected error during import with operation delete for scheduled_updates.scheduled_update_type.node_embargo: Attempt to create a field body that does not exist
> on entity type node.
> Unexpected error during import with operation delete for scheduled_updates.scheduled_update_type.multiple_node_embargo: Attempt to create a field body that does
> not exist on entity type node.
> Unexpected error during import with operation delete for contact.form.sitewide: Attempt to create a field body that does not exist on entity type node.

To try and pin down where these configurations are coming from, I used `scheduled_update` as my needle, and my /config/default files as my haystack.  I have no removed all files with reference to `scheduled_update` (understanding I'll need to manually re-configure scheduled updates after I get this all sorted out), but I am still getting the error above when running `blt setup`.  Looking in the Lightning install profile, I see config files in the install profiles that I believe would generate the errors above, but since they are part of Lightning I am not sure of the most stable way to override them.  Any suggestions?

**Comment by balsama** [comment]
Just noticed this line in your original issue:

> I am building Lightning via composer: `"drupal/lightning": "^2.13"`

The drupal namespace will point you to the D.O packagist which doesn't support distributions (you won't get any of Lightning's dependencies). You need to pull lightning from the main packagist. We namespace it with acquia there and use proper semver tags. So try changing that line to:

`"acquia/lightning": "^2.1.3"`

**Comment by gabe-connolly** [comment]
Thanks for following up, @balsama .  Switching over to the acquia/lightning package fixed the `plugin does not exist` error I was seeing.  I was also finally able to resolve the config import issues by manually copying over the config files from the lightning module install profiles.  I suspect that somewhere earlier in the project, some of the lightning modules were incorrectly removed which resulted in corrupted config being stored in our repo.  Once the corresponding config settings were copied into our /config/default directory, the config import that is run as part of `blt setup` was able to complete without errors.

**Comment by mattferderer** [comment]
@gabe-connolly where did you find the patch that helped you fix the "entity_browser_widget_context plugin does not exist issue"?

The current composer.json file that I see looks like this:

```json
"drupal/entity_browser": {
    "2865928 - The View widget should filter based on field settings":
    "https://www.drupal.org/files/issues/entity-browser-view-context-2865928-8.patch",
    "2877751 - Inform users how many items they can add to a field that uses an entity browser":
    "https://www.drupal.org/files/issues/2877751-10.patch",
}
```
I'm curious as I'm having the same issue as you were. I can file it in a separate issue though.

**Comment by gabe-connolly** [comment]
@mattferderer the patch I was looking for is already bundled in with "acquia/lightning": "^2.1.3".  Once I updated my composer to use that distribution, that fixed my `plugin does not exist` issue.  Before I updated my composer.json file, I had been using drupal/lightning instead of acquia/lightning, so the patch wasn't including in what I was pulling in.

Hope that helps!

**Comment by dman-coders** [comment]
Related - I was getting the same issue, from a mostly-vanilla, but non-blt project that was including 
"acquia/lightning": "^2.1"
as well as 
"cweagans/composer-patches": "~1.0",
and other needful things.

It turned out that I did not, however have 
`    "extra": {
        "enable-patching": true,
`
in the top-most composer.json.

This seems necessary: "Allowing patches to be applied from dependencies" https://github.com/cweagans/composer-patches 

Thanks to hints above, I was able to compare a pure-blt composer.json with the one I had (from another source - platform.sh) and spot the difference.

After doing that, I had to delete and re-install /profiles/contrib/lightning and half of /modules/contrib to get the patching to kick in again.

So from my experience - this error was not about entity_browser specifically, it's just one of the things that goes wrong if your composer failed to patch things automatically.
